src/main/java/io/supertokens/config/Config.java

```diff
@@ -57,9 +58,16 @@ private Config(Main main, String configFilePath) throws InvalidConfigException,
     private Config(Main main, JsonObject jsonConfig) throws IOException, InvalidConfigException {
         this.main = main;
         final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
-        CoreConfig config = mapper.readValue(jsonConfig.toString(), CoreConfig.class);
-        config.normalizeAndValidate(main);
-        this.core = config;
+        try {
+            CoreConfig config = mapper.readValue(jsonConfig.toString(), CoreConfig.class);
```

Comment:
as discussed, we need to disallow converstion of int to bool that happens implicitly for yaml.